syntax/parse.go

```diff
@@ -66,16 +72,25 @@ func ParseExpr(filename string, src interface{}) (expr Expr, err error) {
 }
 
 type parser struct {
-	in     *scanner
-	tok    Token
-	tokval tokenValue
+	in       *scanner
+	tok      Token
+	tokval   tokenValue
+	comments []Comment
 }
 
 // nextToken advances the scanner and returns the position of the
 // previous token.
 func (p *parser) nextToken() Position {
 	oldpos := p.tokval.pos
 	p.tok = p.in.nextToken(&p.tokval)
+	// save comments
+	for p.tok == LINE_COMMENT || p.tok == SUFFIX_COMMENT {
+		if keepComments {
```

Comment:
Pass the comments flag down into the scanner, so that p.in.nextToken never returns comment tokens unless that mode is enabled.